### ISSUE TITLE
AWS::Kinesis::Stream::StreamModeDetails isn't identified as part of the resource schema

### ISSUE THREAD

**Issue by ohadbenita** [aws-cloudformation/cfn-lint#2203]
*cfn-lint version: 0.56.3*

When trying to add StreamModeDetails to AWS::Kinesis::Stream cfn-lint doesn't identify the field and fails validation.

Consider the following template:
```yaml
Resources:
  MyStream:
    Type: AWS::Kinesis::Stream
    DeletionPolicy: Delete
    Properties:
      Name: my-stream-name
      RetentionPeriodHours: 12
      StreamModeDetails:
        StreamMode: ON_DEMAND
```
According to the CloudFormation documentation, this is a valid template but it fails linting by the latest version of cfn-lint.

Any help would be appreciated.

**Comment by PatMyron** [comment]
added in https://github.com/aws-cloudformation/cfn-lint/pull/2188

upgrade to the latest version of `cfn-lint`

**Comment by ohadbenita** [comment]
Thanks, you are completely right.
